spec.html

```diff
@@ -28276,130 +28172,88 @@ <h1>Math.imul ( _x_, _y_ )</h1>
-          </li>
-        </ul>
+        <emu-alg>
+          1. Let _n_ be ? ToNumber(_x_).
+          1. If _n_ is *NaN* or _n_ &lt; -1, return *NaN*.
+          1. If _n_ is *+0*, _n_ is *-0*, or _n_ is *+&infin;*, return _n_.
+          1. If _n_ is -1, return *-&infin;*.
+          1. Return an implementation-approximated value representing the result of the natural logarithm of 1 + _n_.
+        </emu-alg>
       </emu-clause>
 
       <emu-clause id="sec-math.log10">
         <h1>Math.log10 ( _x_ )</h1>
         <p>Returns an implementation-approximated value representing the result of the base 10 logarithm of _x_.</p>
-        <ul>
-          <li>
-            If _x_ is *NaN*, the result is *NaN*.
-          </li>
-          <li>
-            If _x_ is less than 0, the result is *NaN*.
-          </li>
-          <li>
-            If _x_ is *+0*, the result is *-&infin;*.
-          </li>
-          <li>
-            If _x_ is *-0*, the result is *-&infin;*.
-          </li>
-          <li>
-            If _x_ is 1, the result is *+0*.
-          </li>
-          <li>
-            If _x_ is *+&infin;*, the result is *+&infin;*.
-          </li>
-        </ul>
+        <emu-alg>
+          1. Let _n_ be ? ToNumber(_x_).
+          1. If _n_ is *NaN* or _n_ &lt; 0, return *NaN*.
+          1. If _n_ is *+0* or _n_ is *-0*, return *-&infin;*.
+          1. If _n_ is 1, return *+0*.
+          1. If _n_ is *+&infin;*, return *+&infin;*.
+          1. Return an implementation-approximated value representing the result of the base 10 logarithm of _n_.
+        </emu-alg>
       </emu-clause>
 
       <emu-clause id="sec-math.log2">
         <h1>Math.log2 ( _x_ )</h1>
         <p>Returns an implementation-approximated value representing the result of the base 2 logarithm of _x_.</p>
-        <ul>
-          <li>
-            If _x_ is *NaN*, the result is *NaN*.
-          </li>
-          <li>
-            If _x_ is less than 0, the result is *NaN*.
-          </li>
-          <li>
-            If _x_ is *+0*, the result is *-&infin;*.
-          </li>
-          <li>
-            If _x_ is *-0*, the result is *-&infin;*.
-          </li>
-          <li>
-            If _x_ is 1, the result is *+0*.
-          </li>
-          <li>
-            If _x_ is *+&infin;*, the result is *+&infin;*.
-          </li>
-        </ul>
+        <emu-alg>
+          1. Let _n_ be ? ToNumber(_x_).
+          1. If _n_ is *NaN* or _n_ &lt; 0, return *NaN*.
+          1. If _n_ is *+0* or _n_ is *-0*, return *-&infin;*.
+          1. If _n_ is 1, return *+0*.
+          1. If _n_ is *+&infin;*, return *+&infin;*.
+          1. Return an implementation-approximated value representing the result of the base 2 logarithm of _n_.
+        </emu-alg>
       </emu-clause>
 
       <emu-clause id="sec-math.max">
-        <h1>Math.max ( _value1_, _value2_, ..._values_ )</h1>
+        <h1>Math.max ( ..._values_ )</h1>
```

Comment:
It may be surprising, but this change is actually normative. `Math.max.length` is defined to be `2` because of the number of mandatory parameters listed here. We can resolve this by explicitly stating that the length of this function is `2` in the description. Or you could change the signature back. Either way works for me.

Reply:
I have a strong preference to change the signature back, and to minimize the places in the spec where the length is explicitly overriding the signature.

Reply:
Agreed!